### PR TITLE
[multi-device] Bugfixes and throttle pairing check

### PR DESCRIPTION
**Pull request by BeaudanBrown**
First batch of fixes for bugs found when testing MD
Brief descriptions in commit messages
Also put a simple timestamp guard around the device pairing checks

### CODE REVIEW

File: libtextsecure/contacts_parser.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ ProtoParser.prototype = {
       if (this.buffer.limit === this.buffer.offset) {
         return undefined; // eof
       }
-      const len = this.buffer.readVarint32();
+      const len = this.buffer.readInt32();
```
Comment (sachaaaaa) [inline libtextsecure/contacts_parser.js]: did this break on mobile?

Comment (BeaudanBrown) [reply libtextsecure/contacts_parser.js]: Yes this broke iOS I believe, this seems to work on all platforms now tho

Comment (sachaaaaa) [reply libtextsecure/contacts_parser.js]: Nice catch

Comment (BeaudanBrown) [reply libtextsecure/contacts_parser.js]: Oh no it was Niels/Mik, I was just following orders :nerd_face:

File: js/models/conversations.js [hunk 2]
```diff
@@ -766,9 +766,15 @@
     isSecondaryDevice() {
       return !!this.get('secondaryStatus');
     },
-    async setSecondaryStatus(newStatus) {
+    getPrimaryDevicePubKey() {
+      return this.get('getPrimaryDevicePubKey') || this.id;
```
Comment (sachaaaaa) [inline js/models/conversations.js]: Typo?
```suggestion
      return this.get('primaryDevicePubKey') || this.id;
```

Comment (BeaudanBrown) [reply js/models/conversations.js]: Lol yes

**Pull request merged**